test/unit/sql/cte.test.js

```diff
@@ -0,0 +1,313 @@
+        
+        expectsql(sql.selectQuery(User.getTableName(), options, User), expected);
+      };
+
+
+      it('automatically connects to single CTE', function () {
+
+        selectTest(
+          {
+            cte: [{
+              name: 'a',
+              initial: { where: { username: 'user3' } },
+              recursive: { next: 'report' }
+            }]
+          },
+          {
+            sqlite: 'WITH RECURSIVE a( `id`,`amount`,`createdAt`,`updatedAt`,`userId` ) AS ( SELECT `id`, `amount`, `createdAt`, `updatedAt`, `userId` FROM `users` AS `user` WHERE `user`.`username` = \'user3\' UNION SELECT `report`.`id`, `report`.`amount`, `report`.`createdAt`, `report`.`updatedAt`, `report`.`userId` FROM `a` INNER JOIN `users` AS `report` ON `a`.`id` = `report`.`userId`  ) SELECT `user`.* FROM `users` AS `user` INNER JOIN `a` ON `user`.`id` = `a`.`id`;',
+            postgres: 'WITH RECURSIVE a( "id","amount","createdAt","updatedAt","userId" ) AS ( SELECT "id", "amount", "createdAt", "updatedAt", "userId" FROM "users" AS "user" WHERE "user"."username" = \'user3\' UNION SELECT "report"."id", "report"."amount", "report"."createdAt", "report"."updatedAt", "report"."userId" FROM "a" INNER JOIN "users" AS "report" ON "a"."id" = "report"."userId"  ) SELECT "user".* FROM "users" AS "user" INNER JOIN "a" ON "user"."id" = "a"."id";'
+          });
+
+      });
+      
+      it('correctly adds additional attributes to CTE', function () {
+
+        selectTest(
+          {
+          cte: [{
+            name: 'a',
+            cteAttributes: ['total'],
+            initial: {
+              total: { $model: 'amount' },
+              where: { username: 'user3' }
+            },
+            recursive: { 
+              next: 'report',
+              total: { $add: [{ $cte: 'total'}, {$model: 'amount'}]} 
+            }
+          }]
+        },
+        {
+          sqlite: 'WITH RECURSIVE a( `id`,`amount`,`createdAt`,`updatedAt`,`userId`,`total` ) AS ( SELECT `id`, `amount`, `createdAt`, `updatedAt`, `userId`, `user`.`amount` FROM `users` AS `user` WHERE `user`.`username` = \'user3\' UNION SELECT `report`.`id`, `report`.`amount`, `report`.`createdAt`, `report`.`updatedAt`, `report`.`userId`, (`a`.`total` + `report`.`amount`) FROM `a` INNER JOIN `users` AS `report` ON `a`.`id` = `report`.`userId`  ) SELECT `user`.* FROM `users` AS `user` INNER JOIN `a` ON `user`.`id` = `a`.`id`;',
+          postgres: 'WITH RECURSIVE a( "id","amount","createdAt","updatedAt","userId","total" ) AS ( SELECT "id", "amount", "createdAt", "updatedAt", "userId", "user"."amount" FROM "users" AS "user" WHERE "user"."username" = \'user3\' UNION SELECT "report"."id", "report"."amount", "report"."createdAt", "report"."updatedAt", "report"."userId", ("a"."total" + "report"."amount") FROM "a" INNER JOIN "users" AS "report" ON "a"."id" = "report"."userId"  ) SELECT "user".* FROM "users" AS "user" INNER JOIN "a" ON "user"."id" = "a"."id";'
```

Comment:
You can use `default` and replace ticks with [ and ] - that avoids duplicating expected queries when the only difference it the ticks